packages/language-server/src/manifest-handling.ts

```diff
@@ -0,0 +1,100 @@
+import { maxBy, map, filter } from "lodash";
+import { readFile } from "fs-extra";
+import { URI } from "vscode-uri";
+import { dirname } from "path";
+import globby from "globby";
+import { FileChangeType } from "vscode-languageserver";
+
+type AbsolutePath = string;
+type ManifestData = Record<AbsolutePath, { isFlexEnabled: boolean }>;
+const manifestData: ManifestData = Object.create(null);
+let manifestDocuments: string[];
+
+export function isManifestDoc(uri: string): boolean {
+  return uri.endsWith("manifest.json");
+}
+
+export async function initializeManifestDocuments(
+  workspaceFolderPath: string
+): Promise<void[]> {
+  manifestDocuments = await findAllManifestDocumentsInWorkspace(
+    workspaceFolderPath
+  );
+  const readManifestPromises = map(manifestDocuments, async (manifestDoc) => {
+    const isFlexEnabled = await readFlexEnabledFlagFromManifestFile(
+      manifestDoc
+    );
+    // Parsing of manifest.json failed because the file is invalid
```

Comment:
the comment describes what happened (failure to parse).
But not why we ignore this scenario, because we want to keep the last valid state of the manifest (e.g manifest.json being live edited).